kubetest2/kubetest2-kind/deployer/deployer.go

```diff
@@ -26,44 +35,154 @@ const Name = "kind"
 
 // Usage returns the usage for the deployer
 func Usage() string {
-	return "  NONE - kind takes no args currently.\n"
+	return bindFlags(&deployer{}).FlagUsages()
```

Comment:
this pattern already suggests that the kubetest2 flags passed to the deployer don't need to be marked hidden if it is replicated, but I don't think that will hurt.

we could not pass in those flags, but then the deployer would not be able to error on unknown flags.

Reply:
yeah, I've been discussing this with @amwat, especially if deployers _only_ were allowed to have flags (not args) we could do this instead:

Reply:
```
type NewDeployer func(common Options) (Deployer, flags *pflag.FlagSet)
```
- kubetest2 creates the deployer, gets back flags
- deployer flags are printed seperately in help output still
- deployers don't need to implement usage or help output
- don't need to pass in hidden flags
- flag-less deployers just return nil

Reply:
the deployer returns flags now, this is much simpler if every so slightly less flexible